src/datasources/errors/helpers/is-unique-constraint-error.helper.ts

```diff
@@ -0,0 +1,13 @@
+import { QueryFailedError } from 'typeorm';
+
+const UNIQUE_CONSTRAINT_ERROR_CODE = '23505';
+
+export function isUniqueConstraintError(
+  err: unknown,
+): err is QueryFailedError & { driverError: { code: string; detail: string } } {
```

Comment:
Nit: `driverError` extends an `Error` so we can't be certain that `detail` exists:

```suggestion
): err is QueryFailedError & {
  driverError: { code: typeof UNIQUE_CONSTRAINT_ERROR_CODE; detail?: string };
} {
```